src/compiler/checker.ts

```diff
@@ -1552,24 +1560,30 @@ namespace ts {
          * Not passing `lookupTable` and `exportNode` disables this collection, and just extends the tables
          */
         function extendExportSymbols(target: SymbolTable, source: SymbolTable, lookupTable?: Map<ExportCollisionTracker>, exportNode?: ExportDeclaration) {
-            for (const id in source) {
-                if (id !== "default" && !target[id]) {
-                    target[id] = source[id];
+            if (!source) return;
+
+            source.forEach((sourceSymbol, id) => {

```

Comment:
Maybe use `source && source.forEach(...)` and get rid of the `if (!source) return` above.